src/lib/components/DeckGLMap/layers/piechart/pieChartLayer.ts

```diff
@@ -0,0 +1,96 @@
+import { CompositeLayer, COORDINATE_SYSTEM } from "@deck.gl/core";
+import { GeoJsonLayer, GeoJsonLayerProps } from "@deck.gl/layers";
+import { CompositeLayerProps } from "@deck.gl/core/lib/composite-layer";
+import { Feature } from "geojson";
+
+export interface PieChartLayerProps<D> extends CompositeLayerProps<D> {
+    pieRadius: number;
+}
+
+const defaultProps = {};
+export default class PieChartLayer extends CompositeLayer<
+    Feature,
+    PieChartLayerProps<Feature>
+> {
+    renderLayers(): GeoJsonLayer<Feature>[] {
```

Comment:
You can get the viewport parameters from the layer context (`this.context.viewport`) and use that for computing the radius.
Check the viewport parameters here: https://deck.gl/docs/api-reference/core/viewport